app/scripts/modules/netflix/fastProperties/view/rollouts/FastPropertyRollouts.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from 'react';
+import { get, groupBy } from 'lodash';
+import { Subject } from 'rxjs/Subject';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { Application } from 'core/application/application.model';
+import { ApplicationDataSource } from 'core/application/service/applicationDataSource';
+import { IExecution } from 'core/domain';
+import { Execution } from 'core/delivery/executionGroup/execution/Execution';
+import { Sticky } from 'core/utils/stickyHeader/Sticky';
+import { collapsibleSectionStateCache } from 'core/cache/collapsibleSectionStateCache';
+import { $state } from 'core/uirouter';
+import { IFilterTag } from 'core/filterModel/FilterTags';
+
+import './FastPropertyRollouts.less';
+
+interface IProps {
+  application: Application,
+  filters: IFilterTag[],
+  filtersUpdatedStream: Subject<IFilterTag[]>,
+}
+
+interface IState {
+  open: boolean;
+  filteredExecutions: IExecution[];
+  loading: boolean;
+  loadError: boolean;
+}
+
+@autoBindMethods
+export class FastPropertyRollouts extends React.Component<IProps, IState> {
+  private dataSourceUnsubscribe: () => any;
+  private dataSource: ApplicationDataSource;
+  private runningDataSource: ApplicationDataSource;
+  private isGlobal: boolean;
+
+  constructor(props: IProps) {
+    super(props);
+    const sectionCacheKey = this.getSectionCacheKey();
+    this.isGlobal = this.props.application.global;
+    this.state = {
+      open: this.isGlobal || !collapsibleSectionStateCache.isSet(sectionCacheKey) || collapsibleSectionStateCache.isExpanded(sectionCacheKey),
+      filteredExecutions: [],
+      loading: true,
+      loadError: false,
+    };
+    this.dataSource = props.application.getDataSource('propertyPromotions');
+    this.runningDataSource = props.application.getDataSource('runningPropertyPromotions');
+    this.props.filtersUpdatedStream.subscribe(() => this.filterExecutions());
+  }
+
+  public componentDidMount(): void {
+    this.dataSource.activate();
+    this.dataSourceUnsubscribe = this.dataSource.onRefresh(null, () => this.filterExecutions());
+    this.dataSource.ready().then(
+      () => this.setState({loading: false, loadError: false}),
+      () => this.setState({loading: false, loadError: true})
+    );
+    if (!this.state.open && $state.current.name.endsWith('.execution')) {
+      this.setState({open: true});
+    }
+  }
+
+  public componentWillUnmount(): void {
+    this.dataSource.deactivate();
+    if (this.dataSourceUnsubscribe) {
+      this.dataSourceUnsubscribe();
+    }
+  }
+
+  public componentWillReceiveProps(): void {
+    this.filterExecutions();
+  }
+
+  private toggle(): void {
+    const open = !this.state.open;
+    this.setState({open});
```

Comment:
Doing a toggle this way (get state, set state as inverse) can sometimes cause weird toggling behavior when clicking the toggle rapidly since `setState()` is async.